File: Springboot/src/main/java/com/playarena/service/ProductService.java
package com.playarena.service;

import com.playarena.entity.Product;
import com.playarena.repository.ProductRepository;
import org.springframework.stereotype.Service;
import java.util.List;

@Service
public class ProductService {

    private final ProductRepository productRepository;

    public ProductService(ProductRepository productRepository) {
        this.productRepository = productRepository;
    }

    public Product addProduct(Product product) {
        return productRepository.save(product);
    }

    public List<Product> getAllProducts() {
        return productRepository.findAll();
    }

    public Product getProductById(Long id) {
        return productRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Product not found"));
    }

    public Product updateProduct(Long id, Product product) {
        Product existing = getProductById(id);

        existing.setName(product.getName());
        existing.setCategory(product.getCategory());
        existing.setPrice(product.getPrice());
        existing.setDescription(product.getDescription());
        existing.setImageUrl(product.getImageUrl());
        existing.setStock(product.getStock());

        return productRepository.save(existing);
    }

    public void deleteProduct(Long id) {
        productRepository.deleteById(id);
    }
}

File: PlayArena/src/services/productService.js
import API from "./api";

export const getAllProducts = async () => {
  return await API.get("/products");
};

export const getProductById = async (id) => {
  return await API.get(`/products/${id}`);
};

export const addProduct = async (productData) => {
  return await API.post("/admin/products", productData);
};

export const updateProduct = async (id, productData) => {
  return await API.put(`/admin/products/${id}`, productData);
};

export const deleteProduct = async (id) => {
  return await API.delete(`/admin/products/${id}`);
};